src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -804,42 +869,54 @@ private void CopyChangedNestedProperties(TStructuralType original)
                 dynamic deltaNestedResource = _deltaNestedResources[nestedResourceName];
                 dynamic originalNestedResource = null;
 
-                if (!TryGetPropertyRef(original, nestedResourceName, out originalNestedResource))
-                {
-                    throw Error.Argument(nestedResourceName, SRResources.DeltaNestedResourceNameNotFound,
-                        nestedResourceName, original.GetType());
-                }
-
-                if (originalNestedResource == null)
+                if (deltaNestedResource is IDeltaSet)
                 {
-                    // When patching original target of null value, directly set nested resource.
-                    dynamic instance = deltaNestedResource.GetInstance();
-
-                    // Recursively patch up the instance with the nested resources.
-                    deltaNestedResource.CopyChangedValues(instance);
+                    IODataAPIHandler apiHandlerNested = apiHandler.GetNestedHandler(original, nestedResourceName);
```

Comment:
Default value for `apiHandler` parameter is null. Shouldn't we confirm that it's not null before invoking `GetNestedHandler` method?

Reply:
Added the null check